grp/simple.gi

```diff
@@ -887,9 +887,90 @@ local id;
+  e:=LogInt(q,p);
+  if class="L" and n=2 then
+    f:=FreeGroup("d","f");
+    rels:=StringFormatted("d{}=f{}=[d,f]=1",Gcd(q-1,2),e);
+  elif class="L" and n>=3 then
+    f:=FreeGroup("d","g","f");
+    rels:=StringFormatted("d{}=g2=f{}=[g,f]=1,d^g=D,d^f=d{}",Gcd(q-1,n),e,p);
+  elif class="U" and n>=3 then
+    f:=FreeGroup("d","g","f");
+    rels:=StringFormatted("d{}=g2=1,f{}=g,d^g=D,d^f=d{}",Gcd(q+1,n),e,p);
+  elif class="S" and n>=2 and (n<>4 or p<>2) then
+    f:=FreeGroup("d","f");
+    rels:=StringFormatted("d{}=f{}=[d,f]=1",Gcd(q-1,2),e);
+  elif class="S" and n=4 and p=2 then
+    f:=FreeGroup("g","f");
+    rels:=StringFormatted("g2=f,f{}=1",e);
+  elif (class="O" or class="O0" or class="OO") and n>=3 then
+    # special case of characteristic 2 -- copy S status
+    if p=2 then
+      f:=FreeGroup("d","f");
+      rels:=StringFormatted("d{}=f{}=[d,f]=1",Gcd(q-1,2),e);
+    else
+      f:=FreeGroup("d","f");
+      rels:=StringFormatted("d2=f{}=[d,f]=1",e);
+    fi;
+  elif class="O+" and n>=6 and IsEvenInt(n) and n<>8 and p=2 then
+    f:=FreeGroup("g","f");
+    rels:=StringFormatted("g2=f{}=[g,f]=1",e);
+  elif class="O+" and n=8 and p=2 then
+    f:=FreeGroup("t","g","f");
+    rels:=StringFormatted("t^3=g2=(gt)2=f{}=[t,f]=[g,f]=1",e);
+  elif class="O-" and n>=4 and IsEvenInt(n) and p=2 then
+    f:=FreeGroup("g","f");
+    rels:=StringFormatted("g2,f{}=g",e);
+  elif class="O+" and n=8 and IsOddInt(p) then
+    f:=FreeGroup("p","t","g","d","f");
+    rels:=StringFormatted(
+      "p2=t3=g2=(gt)2=d2=1,d^t=p,p^dp,(dg)2=p,f{}=[d,f]=[t,f]=[g,f]=1",e);
+  elif class="O+" and n>=12 and n mod 4=0 and IsOddInt(q) then
+    f:=FreeGroup("p","g","d","f");
+    rels:=StringFormatted("p2=g2=d2=1,(dg)2=p,f{}=[d,f]=[g,f]=1",e);
+  elif class="O+" and n>=6 and n mod 4=2 and q mod 4=1 then
+    f:=FreeGroup("p","g","d","f");
+    rels:=StringFormatted("p2=g2=1,d2=p,d^g=D,f{}=[g,f]=1,d^f=d{}",e,p);
+  elif class="O+" and n>=6 and n mod 4=2 and q mod 4=3 then
+    f:=FreeGroup("g","d","f");
+    rels:=StringFormatted("g2=d2=[d,g]=f{}=[g,f]=[d,f]=1",e);
+  elif class="O-" and n>=4 and (n mod 4=0 or q mod 4=1) and IsOddInt(q) then
+    f:=FreeGroup("g","d","f");
+    rels:=StringFormatted("g2=d2=[d,g]=[d,f]=1,f{}=g",e);
+  elif class="O-" and n>=4 and n mod 4=2 and q mod 4=3 then
+    f:=FreeGroup("p","g","d","f");
+    rels:=StringFormatted("p2=g2=1,d2=p,d^g=D,f{}=[g,f]=[d,f]=1",e);
+  else
+    return fail;
+  fi;
+  gp:=f/ParseRelators(f,rels);
+  SetReducedMultiplication(gp);
+  Size(gp);
+  return gp;
+end);
+
 InstallOtherMethod(DataAboutSimpleGroup,true,[IsRecord],0,
 function(id)
-local nam,e,EFactors,par,expo,prime,result,aut,i;
+local
+nam,e,EFactors,par,expo,prime,result,aut,i,classical,classaut,shortname;
+
+  shortname:=function(gp)
+  local s;
+    if IsCyclic(gp) then
+      return String(Size(gp));
+    elif Size(gp)<=31 or Size(gp) in [33..47] then
+      s:=StructureDescription(gp);
+      s:=Filtered(s,x->not x in "C ");
```

Comment:
What exactly is this test about? It seems to remove all spaces and all C from the structure description? The result won't be in stringrep, though. How about this instead?
```suggestion
      s:=RemoveCharacters(s, "C ");
```

Reply:
This is about producing names for subgroups of outer automorphism groups that is consistent with what is used in the table of marks library (to find the group there). E.g. `StructureDescription` gives `C2 x C2`, while the table of marks (and character table) libraries use .2x2